lib/src/file_util.rs

```diff
@@ -57,6 +57,14 @@ pub fn create_or_reuse_dir(dirname: &Path) -> io::Result<()> {
     }
 }
 
+/// Creates a directory (including any intermediate directories) or does nothing
+/// if the directory already exists.
+///
+/// Returns the underlying error if any directory can't be created.
+pub fn create_dir_all(dirname: &Path) -> io::Result<()> {
+    fs::create_dir_all(dirname)
```

Comment:
maybe you forgot to extract `create_parent_dirs()` from local_working_copy.rs?

(and it's preferred to do that in separate commit)

Reply:
I didn't realize what you were suggesting.  The error types are different so the entire thing is a bit awkward.  I will do it in a separate commit.

Reply:
Indeed. `create_parent_dirs()` will have to return `io::Error` or something.

Can you remove `file_util::create_dir_all()` then? It doesn't make sense to wrap std fn. The function interface will be different anyway (because `create_parent_dirs()` ensures that no directories are created outside of the base directory.)

Reply:
Done